src/NHibernate/Criterion/ContainsExpression.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Linq;
+using NHibernate.Dialect;
+using NHibernate.Engine;
+using NHibernate.SqlCommand;
+
+namespace NHibernate.Criterion
+{
+	[Serializable]
+	public class ContainsExpression : AbstractCriterion
```

Comment:
This is a very DBMS specific thing. SQLite "match" is not the same thing as SQL Server "contains" (which required enabling full text indexing). Calling it ContainsExpression may lead to unintended use. Perhaps it should be in companion libraries instead.

Reply:
MATCH is used in SQLite to search against the FTS index - see [here](https://www.sqlitetutorial.net/sqlite-full-text-search/) -, so what am I missing?

BTW, the choice of "ContainsExpression" for FTS was not lightly taken: "contains" is used by more major players (e.g. SQL Server, Oracle, IBM Db2) than "match" is (e.g. PostgreSQL, MySQL, SQLite).

Reply:
Can you clarify why we need this file in a first place. We don't need `AbstractCriterion` based classes for other functions. So how this `Contains` is different?

Reply:
I'm just thinking it sounds quite a lot like a standard "string contains", when it's actually a "fulltext contains", and what that means differs between DBMS implementations.

Reply:
@bahusoid For exposing the function in the criteria API.

Reply:
@gliljas Good point, especially to programmers. How about "FullTextExpression" and Restrictions.FullText?

Reply:
> For exposing the function in the criteria API.

And what's wrong with `Projections.SqlFunction`?

Anyway I think I got it. SQL Server `Contains` function can't be used in comparison expressions (`WHERE Contains(prop, 'xx') = 1`). And there is no way in Criteria to generate bare projection/function call in WHERE statement (`WHERE Contains(prop, 'xx')`)

Am I right?

I think it's better to implement class for bare bool expressions (`BoolExpression`?) which would be rendered simply as wrapped projection. So usage in code can look something like:
```C#
Restrictions.Where(
  Projections.Bool(
    Projections.SqlFunction(
      "contains",
      NHibernateUtil.Bool,
      Projections.Property("Name"),
      Projections.Constant("xx")))
```
And we can leave nice wrapping for user.

Reply:
Yes, something along those lines, but based on an adjective, to align with other restrictions, e.g FullTextContains

Reply:
>  e.g FullTextContains

I see no such restriction. `LikeExpression` and `InsensitiveLikeExpression` is the closest I see. And I'm not mistaken they can also be simplified to `SQLFunctionTemplate` + `Projections.SqlFunction` call.

Reply:
Ohh :))) I see.  It wasn't for me about `FullTextContains` :) So just ignore my last message ( I deleted it)

Reply:
@bahusoid Please keep in mind that the FTS functions (e.g. contains, match) have no return value.